docs/diagrams/AddTaskSequenceDiagram.puml

```diff
@@ -7,29 +7,12 @@ participant "x:AddTaskCommand" as AddTaskCommand LOGIC_COLOR
 
 participant ":Model" as Model MODEL_COLOR
 
-participant ":TaskListBook" as TaskListBook MODEL_COLOR_T2
-
-participant ":TaskList" as TaskList MODEL_COLOR_T3
-
-
 [-> AddTaskCommand : execute()
 activate AddTaskCommand
 
 AddTaskCommand -> Model : addTask(task)
 activate Model
 
-Model -> TaskListBook : addTask(task)
-activate TaskListBook
-
-TaskListBook -> TaskList : add(task)
-activate TaskList
-
-TaskList ---> TaskListBook
-deactivate TaskList
-
-TaskListBook ---> Model
-deactivate TaskListBook
-
 Model -> Model : updateFilteredTaskList()
 activate Model
 
```

Comment:
Might want to check this condition:
```java
if (model.hasTask(task)) {
            throw new CommandException(MESSAGE_DUPLICATE_TASK);
}
```

```puml
alt Task does not exist
  AddTaskCommand -> Model: addTask(Task)
  Model --> AddTaskCommand: success message
  AddTaskCommand -> User: success message
else Task already exists
  AddTaskCommand --> User: error message
```

Reply:
actually what do the rest think? I drew this sequence diagram with the idea to outline the ideal case at a high level. So, I didnt account for the error checking part in the sequence diagram. Should we standardise and add the error checking part?